beetmoverscript/task.py

```diff
@@ -17,6 +19,27 @@ def validate_task_schema(context):
     scriptworker.client.validate_json_schema(context.task, task_schema)
 
 
+def validate_task_scopes(context, manifest):
+    # make sure scopes exist and are properly set
+    scopes = context.task['scopes']
+    for scope in scopes:
+        if scope.startswith("project:releng:beetmover:"):
+            signing_cert_name = scope.split(':')[-1]
+            if re.search('^[0-9A-Za-z_-]+$', signing_cert_name) is not None:
+                break
+            log.warning('scope {} is malformed, skipping!'.format(scope))
+    else:
+        log.critical("No beetmover scopes!")
+        sys.exit(3)
+
+    # prevent uncoordination between the bucket supposed to use for beetmove
+    # and the current credentials scopes
+    if (signing_cert_name not in manifest['s3_prefix_dated'] or
+            signing_cert_name not in manifest['s3_prefix_latest']):
```

Comment:
Yeah, this makes perfect sense! If someone would have CI scopes like "project:releng:beetmover:try" and manifests like 's3_prefix_dated' = "pub/mobile/nightly/2017/01/2017-01-try.." and/or 's3_prefix_latest' = "pub/mobile/nightly/2017/01/2017-01-try.." it would move on to pushing to nightly without barfing ...